### ISSUE TITLE
SwAV: Runtime Error: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu

### ISSUE THREAD

**Issue by JeffreyWardman** [PyTorchLightning/lightning-bolts#567]
## 🐛 Bug

For the SwAV class, setting gpus=-1 in Trainer results in sinkhorn algorithm error due to one variable being on CPU and one on GPU, alongside other issues. 

The class checks if self.gpus > 0 when it should really be checking that self.gpus != 0.

Runtime Error: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu

#### Code sample
```
import pytorch_lightning as pl
from pl_bolts.models.self_supervised import SwAV
from pl_bolts.datamodules import STL10DataModule
from pl_bolts.models.self_supervised.swav.transforms import (
    SwAVTrainDataTransform, SwAVEvalDataTransform
)
from pl_bolts.transforms.dataset_normalizations import stl10_normalization

# data
batch_size = 128
dm = STL10DataModule(data_dir='.', batch_size=batch_size)
dm.train_dataloader = dm.train_dataloader_mixed
dm.val_dataloader = dm.val_dataloader_mixed

dm.train_transforms = SwAVTrainDataTransform(
    normalize=stl10_normalization()
)

dm.val_transforms = SwAVEvalDataTransform(
    normalize=stl10_normalization()
)

# model
model = SwAV(
    gpus=-1,
    num_samples=dm.num_unlabeled_samples,
    dataset='stl10',
    batch_size=batch_size
)

# fit
trainer = pl.Trainer(precision=16)
trainer.fit(model)
```

### Expected behavior

No error.

### Environment

 - PyTorch Version (e.g., 1.0): 1.7.1
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version: 3.7
 - CUDA/cuDNN version: 11/8.0.5
 - GPU models and configuration: 4xV100
 - Any other relevant information:

**Comment by tchaton** [comment]
Dear @ananyahjha93,

Any reason he is passing gpus=-1 to the model ?

 ```
# model
model = SwAV(
    gpus=-1,
    num_samples=dm.num_unlabeled_samples,
    dataset='stl10',
    batch_size=batch_size
)

# fit
trainer = pl.Trainer(precision=16)
```

**Comment by Borda** [comment]
seems we have the same problem with BYOL too #588

**Comment by ananyahjha93** [comment]
I don't think ``gpus=-1`` is a valid argument for this setting.

**Comment by JeffreyWardman** [comment]
@ananyahjha93 the documentation states that Trainer should be able to receive `gpus=-1`, so this should therefore be valid for the SwAV module (and any other module).

**Comment by ananyahjha93** [comment]
@JeffreyWardman we do need the exact number of GPUs for self-supervised learning models as we need to compute steps per training epoch for learning rate schedule. In that case the simplest solution is for the user to input the gpu count themselves and hence the -1 argument becomes invalid in this particular setting.

**Comment by JeffreyWardman** [comment]
@ananyahjha93 The GPU count can be easily determined in the backend with a single line to maintain consistency across the library.
